include/Model/Languages.php

```diff
@@ -303,21 +303,27 @@ public function update( $args ) {
 			}
 
 			// Update menus locations.
-			if ( ! empty( $this->options['nav_menus'] ) ) {
-				foreach ( $this->options['nav_menus'] as $theme => $locations ) {
+			if ( ! empty( $this->options->get( 'nav_menus' ) ) ) {
+				$nav_menus = $this->options->get( 'nav_menus' );
+
+				foreach ( $nav_menus as $theme => $locations ) {
 					foreach ( array_keys( $locations ) as $location ) {
-						if ( ! empty( $this->options['nav_menus'][ $theme ][ $location ][ $old_slug ] ) ) {
-							$this->options['nav_menus'][ $theme ][ $location ][ $slug ] = $this->options['nav_menus'][ $theme ][ $location ][ $old_slug ];
-							unset( $this->options['nav_menus'][ $theme ][ $location ][ $old_slug ] );
+						if ( ! empty( $nav_menus[ $theme ][ $location ][ $old_slug ] ) ) {
+							$nav_menus[ $theme ][ $location ][ $slug ] = $nav_menus[ $theme ][ $location ][ $old_slug ];
+							unset( $nav_menus[ $theme ][ $location ][ $old_slug ] );
 						}
 					}
 				}
+
+				$this->options->set( 'nav_menus', $nav_menus );
 			}
 
 			// Update domains.
-			if ( ! empty( $this->options['domains'][ $old_slug ] ) ) {
-				$this->options['domains'][ $slug ] = $this->options['domains'][ $old_slug ];
-				unset( $this->options['domains'][ $old_slug ] );
+			if ( ! empty( $this->options->get( 'domains' )[ $old_slug ] ) ) {
+				$domains = $this->options->get( 'domains' );
```

Comment:
```suggestion
                         $domains = $this->options->get( 'domains' );
			if ( ! empty( $domains ) ) {
```

Reply:
Done in [d78b566](https://github.com/polylang/polylang/pull/1595/commits/d78b566c2ac53d015116da9415cfd409770b70ff).